rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt

```diff
@@ -2037,14 +2039,30 @@ internal fun <T : AbstractDataDefinition> List<T>.removeDuplicatedDataDefinition
             dataDefinitionMap[it.name] = it
             true
         } else {
-            it.require(dataDefinition.type == it.type) {
+            it.require(it.matchType(dataDefinition)) {
                 "Incongruous definitions of ${it.name}: ${dataDefinition.type} vs ${it.type}"
             }
             false
         }
     }
 }
 
+internal fun AbstractDataDefinition.matchType(dataDefinition: AbstractDataDefinition): Boolean {
+    if (dataDefinition.type == this.type) {
```

Comment:
I woud have implemented it more or less like this, can you try and let me know what you think:
```kotlin
private fun AbstractDataDefinition.matchType(dataDefinition: AbstractDataDefinition): Boolean {
    return when {
        dataDefinition.type == this.type -> true
        dataDefinition.elementSize() == this.elementSize() -> {
            dataDefinition.type is StringType && this.type is DataStructureType ||
            dataDefinition.type is DataStructureType && this.type is StringType ||
            dataDefinition.type is BooleanType && this.type is StringType ||
            dataDefinition.type is StringType && this.type is BooleanType
        }
        else -> false
    }
}
```

Reply:
Works fine